src/Handlers/GeneralUiStateHandler.php

```diff
@@ -305,4 +320,48 @@ protected function _prepareLinksConfig($linksConfig)
 
         return $preparedLinksConfig;
     }
+
+    /**
+     * Convert an iterator to an array.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $iterator The array or Traversable object to convert
+     *
+     * @throws InvalidArgumentException if $iterator is not an array or a Traversable object
+     *
+     * @return array Result of iterator to array transform.
+     */
+    protected function _iteratorToArrayRecursive($iterator)
+    {
+        if (!($iterator instanceof stdClass) && !is_array($iterator) && !$iterator instanceof Traversable) {
```

Comment:
During our conversation, we went over this method, and I mentioned several points, but this implementation does not take them into account.

1. Why do this manual checking and throwing, if you have `_normalizeIterable()`?
2. Casting an `stdClass` to `array` is not recursive.
3. `toArray()` is not a method of any of the accepted interface. Also, not documented anywhere, let alone promise of being recursive.